apps/ocas/src/actions.erl

```diff
@@ -86,56 +49,57 @@ is_valid_action(Action) ->
-        ,  <<"update">> => update
-        ,  <<"move">> => move
-        ,  <<"redirect">> => redirect
-        ,  <<"delete">> => delete
-        ,  <<"snapshot">> => snapshot
-        ,  <<"detonate">> => detonate
-        ,  <<"restore">> => restore
-        ,  <<"save">> => save
-        ,  <<"modify">> => modify
-        ,  <<"throttle">> => throttle
-        ,  <<"delay">> => delay
-        ,  <<"substitute">> => substitute
-        ,  <<"copy">> => copy
-        ,  <<"sync">> => sync
-        ,  <<"distill">> => distill
-        ,  <<"augment">> => augment
-        ,  <<"investigate">> => investigate
-        ,  <<"mitigate">> => mitigate_server
-        ,  <<"remediate">> => remediate
-        ,  <<"response">> => response
-        ,  <<"alert">> => alert
-        },
-
-    %% return {true, server to run} if valid action,
-    %%      otherwise return {false, undefined}
+        #{ <<"alert">>    => { act_alert, alert_server}
+         , <<"deny">>     => { act_deny, deny_server }
+         , <<"mitigate">> => { act_mitigate, mitigate_server }
+         , <<"scan">>     => { act_scan, scan_server }
+         ,  <<"allow">>   => allow
+         ,  <<"augment">> => augment
+         ,  <<"cancel">>  => cancel
+         ,  <<"contain">> => contain
+         ,  <<"copy">> => copy
+         ,  <<"delay">> => delay
+         ,  <<"delete">> => delete
+         ,  <<"detonate">> => detonate
+         ,  <<"distill">> => distill
+         ,  <<"get">> => get
+         ,  <<"investigate">> => investigate
+         ,  <<"locate">> => locate
+         ,  <<"modify">> => modify
+         ,  <<"move">> => move
+         ,  <<"notify">> => notify
+         ,  <<"pause">> => pause
+         ,  <<"query">> => query
+         ,  <<"redirect">> => redirect
+         ,  <<"remediate">> => remediate
+         ,  <<"report">> => report
+         ,  <<"response">> => response
+         ,  <<"restart">> => restart
+         ,  <<"restore">> => restore
+         ,  <<"resume">> => resume
+         ,  <<"save">> => save
+         ,  <<"set">> => set
+         ,  <<"snapshot">> => snapshot
+         ,  <<"start">> => start
+         ,  <<"stop">> => stop
+         ,  <<"substitute">> => substitute
+         ,  <<"sync">> => sync
+         ,  <<"throttle">> => throttle
+         ,  <<"update">> => update
+         },
+
+    %% return {true, module/function to run} if valid action,
+    %%      otherwise return {false, { undefined, undefined}
     ActionValid = maps:is_key(Action, ValidActions),
-    ActionValue = maps:get(Action, ValidActions, undefined),
+    DefaultAction = { undefined, undefined },
+    ActionValue = maps:get(Action, ValidActions, DefaultAction),
     { ActionValid, ActionValue }.
 
-spawn_action( ActionServer, Json ) ->
-    lager:info( "Got to spawn_action for ~p: ", [ ActionServer ] ),
+spawn_action( ActionModule, ActionServer, Json ) ->
+    lager:info( "Got to spawn_action for ~p:~p ", [ ActionModule, ActionServer ] ),
```

Comment:
Line 99 is too long:     lager:info( "Got to spawn_action for ~p:~p ", [ ActionModule, ActionServer ] ),.